internal/util/url.go

```diff
@@ -75,11 +76,10 @@ func (f ReadUnmounter) Close() error {
 }
 
 // FetchResourceAsReader returns a ReadCloser to the data at the url specified. The caller is responsible for
-// closing the reader.
-func FetchResourceAsReader(l *log.Logger, u url.URL) (io.ReadCloser, error) {
+// closing the reader. client is only necessary if the url is http(s).
+func FetchResourceAsReader(l *log.Logger, client HttpClient, u url.URL) (io.ReadCloser, error) {
```

Comment:
(moving comment inline)

To add to that idea, it might be easier if you call the function with the logger, url, and a timeout structure (I think we already do this in some places). That way we don't need any new types and it might be easier to read/follow.